lexer/lexer.go

```diff
@@ -276,20 +276,17 @@ func (l *Lexer) NextToken() token.Token {
 		tok.Position = l.position
 		tok.Literal = ""
 	default:
-		if isLetter(l.ch) {
-			tok.Position = l.position
-			tok.Literal = l.readIdentifier()
-			tok.Type = token.LookupIdent(tok.Literal)
-			return tok
-		} else if isDigit(l.ch) {
+		if isDigit(l.ch) {
 			tok.Position = l.position
 			literal, kind := l.readNumber()
 			tok.Type = kind
 			tok.Literal = literal
 			return tok
-		} else {
-			tok = l.newToken(token.ILLEGAL)
 		}
+		tok.Position = l.position
+		tok.Literal = l.readIdentifier()
+		tok.Type = token.LookupIdent(tok.Literal)
+		return tok
```

Comment:
I think this might cause some issues as we removed the code to handle illegal characters (while we're assuming all characters are legal now)